core/object/script_language.cpp

```diff
@@ -708,6 +708,16 @@ bool PlaceHolderScriptInstance::has_method(const StringName &p_method) const {
 	return false;
 }
 
+Variant PlaceHolderScriptInstance::callp(const StringName &p_method, const Variant **p_args, int p_argcount, Callable::CallError &r_error) {
+	r_error.error = Callable::CallError::CALL_ERROR_INVALID_METHOD;
+#if TOOLS_ENABLED
+	if (Engine::get_singleton()->is_editor_hint()) {
+		return String("Attempt to call a method on a placeholder instance. Check if the script has the tool annotation.");
```

Comment:
"tool annotation" is very specific to GDScript, but this could happen in other scripting languages too. It might be better to refer it as "tool mode" which is more general.

Reply:
That's why I didn't use `@tool` so as not to refer to a specific syntax. In my opinion, the term "annotation" can be taken more broadly to include C# attributes and Python decorators. But perhaps other languages ​​have completely different ways of registering global classes, so you're probably right.